### ISSUE TITLE
Source Route Checking

### ISSUE THREAD

**Issue by jeroenpraat** [redmatrix/redmatrix#246]
_From @ghost on October 10, 2014 16:14_

This issue is mostly for the sake of keeping issues in one place - not expecting it to be perfect immediately.  So far it mostly works.

Wall to wall posts (comments) are presently broken - because they have no route.  Don't know if they'll be picked up by the poller or not yet.  They do get stored, and [correction] do get distributed, but get rejected.

And another correction - it seems it's not that straight forwards.  Reproduced this three times, and then once more where it was fine.  So this is a shit bug with nothing to see, but will leave it open as a top level post.

_Copied from original issue: friendica/red#633_

**Comment by jeroenpraat** [comment]
_From @zzottel on October 10, 2014 21:31_

Is it wall-to-wall only? In some posts I checked, when I follow the link to source, I'm seeing more comments than those I see on my hub. (Won't post links here in the public, ask me for examples if you need some.) I don't know if all of them might have been wall-to-wall comments, but I doubt it.

I currently have the feeling that I'm only getting comments on other people's posts if the commenter is on current code and doesn't comment wall-to-wall, but I can't say for sure.

The poller does seem to pull in missing comments, though, happened with wall-to-wall test comments by Martin on a Second Fiddle post.

I've activated logs since an hour or so now, if you need something, say so.

**Comment by jeroenpraat** [comment]
_From @ghost on October 10, 2014 21:57_

It's somewhat complicated.  Only wall to wall posts go missing and get refused.  It's clearly not as straightforward as it first appeared, but it is only wall to wall.  The only other missing comments I've found have been across versions.  I've seen a couple that appear to have missing comments, but actually belong to different delivery chains - that's most noticeable in the discover tab - but none that go missing without an apparent reason.

**Comment by jeroenpraat** [comment]
_From @zzottel on October 10, 2014 22:11_

I'm missing comments by Giaco to a post by James Morgan where I doubt that they were wall-to-wall. But I also got comments by Adam to one of his Channel-One-Threads, but his hub is on September code. But in the logs, I see only one delivery, shouldn't there be two, a second one by Channel One?

How can I detect where this delivery came from? The [sender] array? That would be his hub, not zothub.com.

**Comment by jeroenpraat** [comment]
_From @zzottel on October 10, 2014 22:21_

No, probably unrelated, I just received a comment on that same James' thread by someone else who is not on current code. So just ditch my comments, sorry. :-)

**Comment by jeroenpraat** [comment]
_From @friendica on October 10, 2014 22:25_

It all depends on the route the comments took. In the check-in I noted that there are going to be issues until we get everybody on the same page. This is one of those places where I should've rolled a protocol update, but there's no way know what protocol revision was running in some other part of a complex delivery chain. So we just have to plunge ahead and work through it. Nothing blew up. Looks like it's 90% working. So now it's just finding the edge cases and fixing them. 

The wall-to-wall issue is sorted.

**Comment by jeroenpraat** [comment]
_From @zzottel on October 10, 2014 22:34_

This one is a new thread from a few hours ago when I already was on new code, so I didn't expect problems related to this new stuff. And I guess they probably aren't related, too, as I received this new comment (but the originating hubs of the thread and the new comment are on old code).

**Comment by jeroenpraat** [comment]
_From @friendica on October 11, 2014 2:28_

By the way - the wall-to-wall comment issue didn't have anything to do with the route. It was a redundant relay delivery which I tried to optimise out yesterday, but it turns out there was a reason for the redundant delivery. We needed it for wall-to-wall posts. I do need to go back and optimise it so that it runs if it's wall-to-wall but not run redundantly if we got relayed from elsewhere. For now I just undid yesterday's optimisation.
